src/flags/date.rs

```diff
@@ -83,6 +83,24 @@ impl Configurable<Self> for DateFlag {
             None
         }
     }
+
+    /// Get a potential `DateFlag` variant from the environment.
+    fn from_environment() -> Option<Self> {
+        if let Ok(value) = std::env::var("TIME_STYLE") {
+            match value.as_str() {
+                "full-iso" => Some(Self::Formatted("%F %T.%f %z".into())),
+                "long-iso" | "iso" => Some(Self::Formatted("%F %R".into())),
```

Comment:
`iso` and `local`(`locale`?) seem to deviate from what gnu ls offers. I am not sure if these are things we can achieve with format string. If not, lets skip it.

---

Other than this (except for some minor difference in format string available in [gnu ls](https://www.cyberciti.biz/faq/linux-unix-formatting-dates-for-display/) and [chrono](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html#specifiers) which I think should be fine), I think we are good to go.

Reply:
Hm you're right, `iso` and `locale` don't match GNU `ls` output. Both should be easy enough though, I'll fix those

Reply:
Okay, fixed both in a new commit

Reply:
Actually `iso` and `locale` is a bit more involved if the year is not current year (example below).

```
$ TIME_STYLE='locale' gls -l LICENSE README.md
-rw-r--r-- 1 meain staff 11359 Dec  8  2018 LICENSE
-rw-r--r-- 1 meain staff 10245 Jan 13 15:26 README.md

$ TIME_STYLE='iso' gls -l LICENSE README.md
-rw-r--r-- 1 meain staff 11359 2018-12-08  LICENSE
-rw-r--r-- 1 meain staff 10245 01-13 15:26 README.md
```

Reply:
You're right. I've only tested with files from 2020 and none of them showed the year. Seems like `iso` and `locale` are impossible with a simple format string. Still, with new enum values in `DateFlag` it should be possible.

Reply:
Okay, `iso` and `locale` now look like GNU `ls` as far as I can tell.

Reply:
`iso` changes the output format if it is [six months ago](https://github.com/coreutils/coreutils/blob/06f70d9ee69163fb2b18647963d6d6e81b6bd35d/src/ls.c#L4385) instead of an year. Plus, I think we could use `%e` instead of `%d` in the format string to avoid the leading zero in date for ones less than 10.

`locale` looks like it is a [bit more involved](https://github.com/coreutils/coreutils/blob/06f70d9ee69163fb2b18647963d6d6e81b6bd35d/src/ls.c#L2433). It needs to show the output based on the user's locale. The [libs](https://github.com/Koka/gettext-rs) that that help with this seems to be providing `ffi` bindings to the c lib and so I think we could probably separate it out to a different PR and skip in this.

```
$ LC_ALL='am_ET' TIME_STYLE='locale' gls -l AUTHORS
-rw-r--r-- 1 meain staff 3812 ጃንዩ 13 21:09 AUTHORS

$ LC_ALL='en_US.UTF-8' TIME_STYLE='locale' gls -l AUTHORS
-rw-r--r-- 1 meain staff 3812 Jan 13 21:09 AUTHORS
```